BraintreeUIKit/Models/BTUIKCardType.m

```diff
@@ -246,13 +246,11 @@ - (NSAttributedString *)formatNumber:(NSString *)input {
 #pragma mark - Validation
 
 - (BOOL)validAndNecessarilyCompleteNumber:(NSString *)number {
-    return (number.length == self.validNumberLengths.lastIndex &&
-            ([BTUIKUtil luhnValid:number] || [self.brand isEqualToString:BTUIKLocalizedString(CARD_TYPE_UNION_PAY)]));
+    return (number.length == self.validNumberLengths.lastIndex && [BTUIKUtil luhnValid:number]);
```

Comment:
Is this logic identical to the raw sdk? Should we just pull it from the raw sdk instead?

Reply:
We don't run Luhn validation in the raw sdk level, just at this level for UI feedback.